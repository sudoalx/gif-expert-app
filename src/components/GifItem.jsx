import PropTypes from "prop-types";

const GifItem = ({ title, url }) => {
  return (
    <div className="card">
      <div className="card-content">
        <img src={url} alt={title} />
        <p>{title}</p>
      </div>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifItem;
