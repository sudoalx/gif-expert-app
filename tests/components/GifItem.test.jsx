import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe("Test GifItem component", () => {
  test("should return a card with the Gif and title", () => {
    const title = "Marvel";
    const url = "https://github.com/iron-man.gif";
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should return image with url and alt specified", () => {
    const title = "Marvel";
    const url = "https://github.com/iron-man.gif";
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test("should display the title in the component", () => {
    const title = "Marvel";
    const url = "https://github.com/iron-man.gif";
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
