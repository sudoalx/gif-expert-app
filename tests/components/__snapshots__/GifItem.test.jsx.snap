// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem component should return a card with the Gif and title 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-content"
    >
      <img
        alt="Marvel"
        src="https://github.com/iron-man.gif"
      />
      <p>
        Marvel
      </p>
    </div>
  </div>
</div>
`;
